src/libmongoc/src/mongoc/mongoc-cursor.c

```diff
@@ -1038,6 +1035,10 @@ _mongoc_cursor_run_command (mongoc_cursor_t *cursor,
     * $readPreference.
     */
    cmd_name = _mongoc_get_command_name (command);
+   if (_mongoc_cursor_get_opt_bool (cursor, MONGOC_CURSOR_EXHAUST)) {
```

Comment:
Consider moving up to where the `if (parts.user_query_flags & MONGOC_QUERY_EXHAUST)` used to be on line 1029.